pkg/utils/decoder.go

```diff
@@ -67,28 +67,29 @@ func YAMLToUnstructured(in []byte) (u unstructured.Unstructured, err error) {
 }
 
 // YAMLToUnstructuredSlice converts a raw yaml document into a slice of pointers to Unstructured objects
-func YAMLToUnstructuredSlice(in []byte) ([]*unstructured.Unstructured, error) {
-	u, err := YAMLToUnstructured(in)
-	if err != nil {
-		return []*unstructured.Unstructured{}, err
-	}
-	if u.IsList() {
-		result := []*unstructured.Unstructured{}
-		err = u.EachListItem(func(obj runtime.Object) error {
-			o, ok := obj.(*unstructured.Unstructured)
-			if !ok {
-				kind := obj.GetObjectKind().GroupVersionKind().Kind
-				return fmt.Errorf("invalid resource of Kind %s", kind)
-			}
-			result = append(result, o)
-			return nil
-		})
+func YAMLToUnstructuredSlice(in []byte) (us []*unstructured.Unstructured, err error) {
+	for _, yaml := range splitYAML(in) {
+		var u unstructured.Unstructured
+		u, err = YAMLToUnstructured(yaml)
 		if err != nil {
-			return []*unstructured.Unstructured{}, err
+			// unable to parse properly, bail
+			break
```

Comment:
Any reason not to just return here? Since that's the effect of the break right?
```suggestion
			return us, err
```

If we do this then the error need not be scoped outside of the for loop which I think would probably be better